### PR TITLE
Add RecordRepresentation component

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/controller/record/RecordRepresentation.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+describe('RecordRepresentation', () => {
+    it('should render the record id when not provided on its parent <Resource>', async () => {
+        render(<NoRecordRepresentation />);
+        await screen.findByText('#1');
+    });
+    it('should render the record id when provided as a field name on its parent <Resource>', async () => {
+        render(<StringRecordRepresentation />);
+        await screen.findByText("The Hitchhiker's Guide to the Galaxy");
+    });
+    it('should render the record id when provided as a function on its parent <Resource>', async () => {
```
Comment (slax57) [inline packages/ra-core/src/controller/record/RecordRepresentation.spec.tsx:19]:
```suggestion
    it('should render the record representation when provided as a function on its parent <Resource>', async () => {
```

File: packages/ra-core/src/controller/record/RecordRepresentation.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        render(<FunctionRecordRepresentation />);
+        await screen.findByText(
+            "The Hitchhiker's Guide to the Galaxy by Douglas Adams"
+        );
+    });
+    it('should render the record id when provided as a component on its parent <Resource>', async () => {
```
Comment (slax57) [inline packages/ra-core/src/controller/record/RecordRepresentation.spec.tsx:25]:
```suggestion
    it('should render the record representation when provided as a component on its parent <Resource>', async () => {
```

**Review by slax57 — CHANGES_REQUESTED** [review]

File: docs/Reference.md [hunk 3]
```diff
@@ -127,6 +127,7 @@ title: "Index"
 
 **- R -**
 * [`<RadioButtonGroupInput>`](./RadioButtonGroupInput.md)
+* [`<RecordRepresentation>`](./RecordRepresentation.md)
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/Reference.md:130]: Added

Comment (slax57) [inline docs/Reference.md:130]: you need to add `useRecordRepresentation` in this page too then

File: docs/useGetRecordRepresentation.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+```tsx
+// in src/posts/PostBreadcrumbs.tsx
+import * as React from 'react';
+import { Breadcrumbs, Typography } from '@mui/material';
+import { Link, useGetRecordRepresentation, useRecordContext, useResourceContext } from 'react-admin';
+
+export const PostBreadcrumbs = () => {
+    const record = useRecordContext(props);
+    const resource = useResourceContext(props);
```
Comment (slax57) [inline docs/useGetRecordRepresentation.md:20]:
```suggestion
    const record = useRecordContext();
    const resource = useResourceContext();
```

(no props available in this example)

**Review by slax57 — CHANGES_REQUESTED** [review]

**Review by slax57 — APPROVED** [review]

**Pull request merged**